camayoc/tests/remote/rho/test_scan.py

```diff
@@ -96,21 +96,34 @@ def test_scan(isolated_filesystem, profile):
             known_facts = {}
             for host in cfg['rho']['hosts']:
                 # find the facts for the scanned host we are inspecting
-                if host['ip'] == row['connection.host']:
+                if host['ip'] == row.get('connection.host'):
                     known_facts = host['facts']
                     break
-            for fact in known_facts.keys():
+            missed_facts = known_facts.keys() - row.keys()
+            if (missed_facts != set()):
```

Comment:
Empty lists means False for python:

```pyhton
>>> bool(set())
False
>>> bool(set([1,2,3]))
True
```

With that we can write cleaner code like:

```python
if missed_facts:
    ...
```

Reply:
Got me again!